frontend/src/pages/shipments/ShipmentOffers.tsx

```diff
@@ -0,0 +1,131 @@
+import cx from 'classnames'
+import { FunctionComponent, useMemo } from 'react'
+import { Column, useSortBy, useTable } from 'react-table'
+import Badge from '../../components/Badge'
+import ButtonLink from '../../components/ButtonLink'
+import TableHeader from '../../components/table/TableHeader'
+import {
+  OffersForShipmentQuery,
+  useAllGroupsMinimalQuery,
+  useOffersForShipmentQuery,
+} from '../../types/api-types'
+import { offerCreateRoute } from '../../utils/routes'
+
+interface Props {
+  shipmentId: number
+}
+
+const ShipmentOffers: FunctionComponent<Props> = ({ shipmentId }) => {
+  // Get the list of offers for this shipment
+  const { data } = useOffersForShipmentQuery({
+    variables: { shipmentId },
+  })
+
+  // Get the list of groups
+  const { data: groups } = useAllGroupsMinimalQuery()
+
+  const offers = useMemo(() => data?.listOffers || [], [data])
+
+  const COLUMNS = useMemo(() => {
+    if (groups?.listGroups && data?.listOffers) {
+      const columns: Column<OffersForShipmentQuery['listOffers'][0]>[] = [
+        {
+          Header: 'ID',
+          accessor: 'id',
+        },
+        {
+          Header: 'Sending Group',
+          accessor: (row) =>
+            groups.listGroups.find((group) => group.id === row.sendingGroupId)
+              ?.name,
+        },
+        {
+          Header: 'Status',
+          accessor: 'status',
+          Cell: ({ value }) => <Badge>{value}</Badge>,
+        },
+        {
+          Header: 'Pallets',
+          accessor: (row) => row.pallets.length,
+        },
+        {
+          Header: 'Photos',
+          accessor: (row) => row.photoUris.length,
+        },
+      ]
+
+      return columns
+    }
+
+    return []
```

Comment:
nit (non-blocking): invert the boolean and early return instead of the large `if` body?